### PR TITLE
UI for audio mute, video mute, camera switch, fullscreen and hangup

### PR DESCRIPTION
**Pull request by samdutton**
Uses inline SVGs to avoid slowing down page load. (These are [well supported](http://caniuse.com/#search=inline%20svg).)

Also added:
• `$()` function for `document.querySelector()`
• `hide()` and `show()` functions to add/remove the class name `hidden`
• `activate()` and `deactivate()` to do the same for `active`

Getting this error with jstdPhantom Grunt task, not sure why: 

> Safari: Reset
> Safari: Reset
> ...
> Total 3 tests (Passed: 3; Fails: 0; Errors: 0) (4.00 ms)
>   Safari 534.34 Mac OS: Run 4 tests (Passed: 3; Fails: 0; Errors 1) (4.00 ms)
>     error loading file: /test/js/main.js:60: TypeError: 'null' is not an object (evaluating 'muteAudioSvg.onclick = toggleAudioMute')
> 
> &gt;&gt; Total Passed: 3, Fails: 0
> 
> PhantomJS threw an error:
> Done, without errors.
>

### CODE REVIEW

**Comment by samdutton** [comment]
New year's @juberti ping!

File: README.md [hunk 1]
```diff
@@ -66,7 +66,9 @@ Patches and issues welcome!
 
 ### Video chat ###
 
-[Complete video chat client (based on Google App Engine)](https://apprtc.appspot.com)
+[AppRTC video chat client](https://apprtc.appspot.com)
```
Comment (juberti) [inline README.md]: Keep the "powered by Google App Engine" thing here - this is a continual source of confusion

File: README.md [hunk 2]
```diff
@@ -66,7 +66,9 @@ Patches and issues welcome!
...
+
+[AppRTC parameters](https://apprtc.appspot.com/params.html)
```
Comment (juberti) [inline README.md]: Make it clear these are URL params: "AppRTC URL parameters"

File: index.html [hunk 3]
```diff
@@ -104,7 +110,9 @@ <h3 id="datachannel">RTCDataChannel</h3>
 
       <h3 id="videoChat">Video chat</h3>
 
-      <p><a href="//apprtc.appspot.com">AppRTC video chat client (based on Google App Engine)</a></p>
+      <p><a href="//apprtc.appspot.com">AppRTC video chat client</a></p>
```
Comment (juberti) [inline index.html]: This should match README.md exactly

File: samples/web/content/apprtc/css/main.css [hunk 4]
```diff
@@ -101,28 +101,6 @@ footer > div.active {
   opacity: .8;
 }
 
-@-webkit-keyframes rotate {
```
Comment (juberti) [inline samples/web/content/apprtc/css/main.css]: did you mean to remove this?

File: samples/web/content/apprtc/js/main.js [hunk 5]
```diff
@@ -345,16 +367,62 @@ function displayError(error) {
...
+  element.classList.remove('active');
+}
+
+function IconSet(icon0, icon1){
+  this.icon0 = document.querySelector(icon0);
+  this.icon1 = document.querySelector(icon1);
+}
+
+IconSet.prototype.toggle = function() {
+  if (this.icon0.classList.contains('hidden')){
```
Comment (juberti) [inline samples/web/content/apprtc/js/main.js]: suggest adding a toggle() function so that IconSet.prototype.toggle just becomes { toggle(icon0); toggle(icon1); }

File: samples/web/content/apprtc/css/main.css [hunk 6]
```diff
@@ -185,3 +163,122 @@ label {
   opacity: 1;
   z-index: 1;
 }
+
+@-webkit-keyframes rotate {
```
Comment (juberti) [inline samples/web/content/apprtc/css/main.css]: ping on this - very curious how you resolved this.

File: samples/web/content/apprtc/params.html [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    </section>
+
+    <section>
+
+      <table>
+        <tr><td><a href="https://apprtc.appspot.com?hd=true">hd=true</a></td><td>Use HD camera resolution <a href="http://googlechrome.github.io/webrtc/samples/web/content/getusermedia/resolution/" title="Video constraints demo">constraints</a>, i.e. minWidth: 1280, minHeight: 720</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?stereo=true">stereo=true</a></td><td>Turn on stereo audio</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?debug=loopback">debug=loopback</a></td><td>Connect to yourself, e.g. to test firewalls</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?ts=turn:foo">ts=[turnserver]</a></td><td>Set TURN server different from the default</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?tp=foo">tp=[password]</a></td><td>Set password for the TURN server</td></tr>
```
Comment (juberti) [inline samples/web/content/apprtc/params.html]: tp no longer exists either

File: samples/web/content/apprtc/params.html [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+    </section>
+
+    <section>
+
+      <table>
+        <tr><td><a href="https://apprtc.appspot.com?hd=true">hd=true</a></td><td>Use HD camera resolution <a href="http://googlechrome.github.io/webrtc/samples/web/content/getusermedia/resolution/" title="Video constraints demo">constraints</a>, i.e. minWidth: 1280, minHeight: 720</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?stereo=true">stereo=true</a></td><td>Turn on stereo audio</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?debug=loopback">debug=loopback</a></td><td>Connect to yourself, e.g. to test firewalls</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?ts=turn:foo">ts=[turnserver]</a></td><td>Set TURN server different from the default</td></tr>
```
Comment (juberti) [inline samples/web/content/apprtc/params.html]: the format of ts should now be something like https://turn.example.org - it's a web frontend to a TURN server now

File: samples/web/content/apprtc/js/main.js [hunk 9]
```diff
@@ -345,16 +367,62 @@ function displayError(error) {
...
+
+  if (this.icon1.classList.contains('hidden')){
+    this.icon1.classList.remove('hidden');
+  } else {
+    this.icon1.classList.add('hidden');
+  }
+};
+
+function showIcons() {
+  if (!icons.classList.contains('active')) {
```
Comment (juberti) [inline samples/web/content/apprtc/js/main.js]: an isVisible function would be useful here

File: samples/web/content/apprtc/params.html [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+        <tr><td><a href="https://apprtc.appspot.com?arc=opus/48000">arc=opus/48000</a></td><td>Set preferred audio receive codec Opus at 48kHz</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?dtls=false">dtls=false</a></td><td>Disable <a href="https://en.wikipedia.org/wiki/Datagram_Transport_Layer_Security" title="Wikipedia article: DTLS">Datagram Transport Layer Security</a></td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?dscp=true">dscp=true</a></td><td>Enable DSCP</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?ipv6=true">ipv6=true</a></td><td>Enable IPv6</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?arbr=48000">arbr=[bitrate]</a></td><td>Set audio receive bitrate, kbps</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?asbr=16000">asbr=[bitrate]</a></td><td>Set audio send bitrate</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?vsbr=1000">vsbr=[bitrate]</a></td><td>Set video receive bitrate</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?vrbr=8000">vrbr=[bitrate]</a></td><td>Set video send bitrate</td></tr>
```
Comment (jiayliu) [inline samples/web/content/apprtc/params.html]: missing 'wshost', 'wsport', and 'wstls', for specifying the websocket server host, port, and if tls is used, respectively.

File: samples/web/content/apprtc/params.html [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+    <section>
+
+      <table>
+        <tr><td><a href="https://apprtc.appspot.com?hd=true">hd=true</a></td><td>Use HD camera resolution <a href="http://googlechrome.github.io/webrtc/samples/web/content/getusermedia/resolution/" title="Video constraints demo">constraints</a>, i.e. minWidth: 1280, minHeight: 720</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?stereo=true">stereo=true</a></td><td>Turn on stereo audio</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?debug=loopback">debug=loopback</a></td><td>Connect to yourself, e.g. to test firewalls</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?ts=turn:foo">ts=[turnserver]</a></td><td>Set TURN server different from the default</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?tp=foo">tp=[password]</a></td><td>Set password for the TURN server</td></tr>
+        <tr><td><a href="https://apprtc.appspot.com?audio=true&amp;video=false">audio=true&amp;video=false</a></td><td>Audio only</td></tr>
```
Comment (jiayliu) [inline samples/web/content/apprtc/params.html]: I thought we were planning to generate the list of params through template. It's hard to keep them in sync this way. 
Also, why not put this in a separate CL?

**Comment by juberti** [comment]
At this point I just want to land this so we can avoid mergedowns against the other pending CLs. IOW I just want to check this in Friday; I will make any remaining fixes manually if needed.

File: README.md (hunk 1 above)
Comment (samdutton) [inline README.md]: Changed.

File: README.md (hunk 2 above)
Comment (samdutton) [inline README.md]: Added in next commit.

File: index.html (hunk 3 above)
Comment (samdutton) [inline index.html]: Done.

File: samples/web/content/apprtc/css/main.css (hunk 4 above)
Comment (samdutton) [inline samples/web/content/apprtc/css/main.css]: Yes. This was only used for the camera-switch animation, which I've removed. Rotation of the video elements is done via a transform.

File: samples/web/content/apprtc/params.html (hunk 7 above)
Comment (samdutton) [inline samples/web/content/apprtc/params.html]: Removed.

File: samples/web/content/apprtc/params.html (hunk 8 above)
Comment (samdutton) [inline samples/web/content/apprtc/params.html]: Changed.

File: samples/web/content/apprtc/js/main.js (hunk 9 above)
Comment (samdutton) [inline samples/web/content/apprtc/js/main.js]: Might be worth an `Icon` class with `isVislble()`, `show()`, `hide()` and `toggle()` methods.

I'll have a go at that separately.

**Comment by samdutton** [comment]
@juberti ptal

File: samples/web/content/apprtc/params.html (hunk 10 above)
Comment (samdutton) [inline samples/web/content/apprtc/params.html]: Thanks. 

Could you give values/examples?

**Comment by juberti** [comment]
lgtm (still curious about how you solved the flipped-buttons problem)

**Comment by jiayliu** [comment]
One issue in main.js:

It calls isFullScreen, which is not defined anywhere.
https://github.com/GoogleChrome/webrtc/blob/master/samples/web/content/apprtc/js/main.js#L377

**Comment by samdutton** [comment]
The shim code was meant to be moved to util.js, but somehow I missed that.

Fixed in commit in github.com/GoogleChrome/webrtc/pull/358.
